### DISCUSSION TITLE
Why do we choose CursorHold instead of CursorMoved event?

### DISCUSSION THREAD

**Discussion by ziontee113** [nvim-neotest/neotest#95] [Q&A] [answered]
Hi, I tried out `CursorMoved` event instead of `CursorHold` and the performance seems normal.
With the default `CursorHold` event, there's a big delay until the `summary window` updates, which feels a bit uncomfortable.

Is `CursorHold` event meant to improve performance? Or are there problems related to `CursorMoved`? I'm curious.

**Comment by rcarriga** [comment, marked as answer]
`CursorMoved` would be too common and so yes it would affect performance.

The `CursorHold` event is delayed by the `updatetime` setting, which by default is 4 seconds. You can decrease this but it causes much more writing to your disk (which is probably an SSD) which can is an issue. There are also other issues with it https://github.com/neovim/neovim/issues/12587. Instead you can use https://github.com/antoinemadec/FixCursorHold.nvim and set the time for it to be small without bad side effects.

**Reply by ziontee113** [reply]
Thank you for the answer. I did install `FixCursorHold` but didn't `let g:cursorhold_updatetime = 100` :smile: , now it's much better :+1:
